### ISSUE TITLE
Memory Usage getting high again

### ISSUE THREAD

**Issue by leaderdog** [mylar3/mylar3#294]
Hi Evilhero,

I'm not really sure which update, but it wasn't the last one.  It may have been the one before that, the Memory usage is starting to creep up again.  It used to stay around 200MB (or less) but last night when I checked my Ram usage for a different reason Mylar was over 1GB again.  I ran the update and this morning (8 hours later) it's just over 700MB and slowly climbing.

Mylar Version: python3-dev
-- git (build 880f5ba278aadb7054f1dfd4b788f0343ad09805)
Python Version : 3.7.4
Language : en_CA.cp1252
Windows 10

**Comment by evilhero** [comment]
Is this when you're not doing anything with Mylar whatsoever during that span of 8hrs ( or even just prior to it ) ? There might be a search gone awry again that I'd need to find, but honestly the last time was just me looking at enough issue reports that it clicked what was happening *shrug* Is there any tracebacks / errors in your logs that would indicate that something failed at some point during that timeframe or possibly even just prior ?

**Comment by leaderdog** [comment]
I didn't have debug logging on,  there are no tracebacks over the past 2 days  I only did one thing with Mylar today and that was run a manual post-process. I did check to see if any books were stuck in "loading" only one was, how long or if it just did the refresh for that book is unknown.

Mylar did return to around 350MB of Memory Use on it's own earlier today.  And right now it's even Less 319MB.

As for that 8 hours, I was asleep ;)  And I don't believe I did anything with Mylar last night except to update the program. I just sporadically check to see if updates are ready.

So, may be just a false alarm, or something got stuck and it worked itself out?

**Comment by evilhero** [comment]
Yeah it could be a false alarm - if something got stuck, over the span of 8 hrs it might have time to work itself out in some other way and bypassing whatever stopped it the first time. Like Mylar will use abit more memory when doing certain tasks, and sometimes they could run close to the same time that they would overlap each other in actions - so the 8hr thing working itself in those kinda cases is feasible, since the odds of it happening repeatedly is pretty slim.

I'd keep an eye on it and see. The key in all of this (if you happen to see the memory climb), is to get the logs out as quickly as you can before they get rotated over. Also remembering any action you did in the GUI prior to that helps when it comes to tracing workflows that overlap when they shouldn't. These types are always tough to track down, as it requires a lot of time, logs and tracing.

**Comment by barbequesauce** [comment]
Any news @leaderdog ?

**Comment by leaderdog** [comment]
I just checked and I'm sitting at 1903MB of RAM usage right now.  Yesterday the highest I saw was 1700ishMB and again the day before that it was around 1700MB.  I checked the log for tracebacks.  There's a couple but they shouldn't have caused the issue.  The last one being yesterday.  Only maybe 4 tracebacks in total.

I did at 2 security cameras to the server and it's constantly hogging Memory as well, but it tops out around 1064MB.  

I'm guessing at this point maybe the problem is the fx 8120 in the server is struggling and so the entire pc may be as well... Although everything else is running in it's acceptable memory uses.  I did buy a new cpu for my pc and the server is going to join the Ryzen revolution with a 2700 and 32GB of RAM.  So once that's done hopefully things will be better.

I'll let you know if anything changes.

**Comment by barbequesauce** [comment]
What are your Bragging Rights numbers, out of curiosity? And what providers are you using?

**Comment by leaderdog** [comment]
**Bragging Rights
# of Series you're watching: 6217
# of Series you're watching that are continuing: 203
# of Issues you're watching: 43412
# of Issues you actually have: 42691
... total HD-space being used: 1.74 TB**

providers are althub and geek.  I turned off the others a very long time ago.

**Comment by evilhero** [comment]
If you have the tracebacks can you paste them? They might not seem related, but there's alot of studd that uses the same module cores repeatidly so it's very possible the traceback is causing a lockup on the queue and never releasing it.

**Comment by leaderdog** [comment]
Here's the care package.  Let me know if it didn't work.
https://www.dropbox.com/s/avu46135wgcufw2/carepackage.zip?dl=0
